### ISSUE TITLE
sync.rsync: leverage check=True in subprocess.run

### ISSUE THREAD

**Issue by OpenBagTwo** [OpenBagTwo/EnderChest#124]
Currently, syncing via rsync checks for errors via inspecting stderr:

https://github.com/OpenBagTwo/EnderChest/blob/1f1b596e4ce40c2b29db622cb12031f3fc7ac881/enderchest/sync/rsync.py#L38-L44

That's silly when we could [just use the kwarg `check=True`](https://docs.python.org/3/library/subprocess.html):

> If _check_ is true, and the process exits with a non-zero exit code, a [CalledProcessError](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError) exception will be raised. Attributes of that exception hold the arguments, the exit code, and stdout and stderr if they were captured.

This should have no user impact but will absolutely simplify the code.

**Comment by OpenBagTwo** [comment]
[CalledProcessError](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError) does not subclass `RuntimeError`, and `check=True` swallows any actual error message (you can tease it out from `CalledProcessError.stderr` but if I'm doing that, what's the point?)

Also: this is only used for the _version check_ (`Popen` doesn't have this kwarg).

**tl;dr** this is a #wontfix
